### PR TITLE
[#5729] feat(client-python): Add distribution expression in Python client

### PR DESCRIPTION
**Pull request by SophieTech88**
### What changes were proposed in this pull request?
Implement distributions expression in python client, add unit test.

### Why are the changes needed?
We need to support the  distributions expressions in python client

Fix: #5729

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Need to pass all unit tests.

### CODE REVIEW

File: clients/client-python/gravitino/api/expressions/distributions/distribution.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from gravitino.api.expressions.expression import Expression
+
+
+class Distribution:
```
Comment (tengqm) [inline clients/client-python/gravitino/api/expressions/distributions/distribution.py:21]: If you are mimicking the Java implementation, you may want to make this an ABC class.

**Comment by xunliu** [comment]
@SophieTech88 I help you improve your PR, Please review it.

File: clients/client-python/gravitino/api/expressions/distributions/distribution.py (hunk 1 above)
Comment (xunliu) [reply clients/client-python/gravitino/api/expressions/distributions/distribution.py]: I help @SophieTech88 fix this problem.

File: clients/client-python/gravitino/api/expressions/distributions/distribution.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    """
+    An interface that defines how data is distributed across partitions.
+    """
+
+    @abstractmethod
+    def strategy(self) -> Strategy:
+        """Return the distribution strategy name."""
+
+    @abstractmethod
+    def number(self) -> int:
```
Comment (tengqm) [inline clients/client-python/gravitino/api/expressions/distributions/distribution.py:34]:
```suggestion
    def buckets(self) -> int:
```

Comment (SophieTech88) [reply clients/client-python/gravitino/api/expressions/distributions/distribution.py:34]: Do we need to make the name consist  to the java file?

Comment (xunliu) [reply clients/client-python/gravitino/api/expressions/distributions/distribution.py:34]: @yuqi1129 Can you give some suggestions?

Comment (yuqi1129) [reply clients/client-python/gravitino/api/expressions/distributions/distribution.py:34]: Both are acceptable to me, I would prefer to keep it consistent with that in Java.

Comment (xunliu) [reply clients/client-python/gravitino/api/expressions/distributions/distribution.py:34]: If we need to modify, I think we can do this in another PR to modify function name in the Python and Java client at same time.

**Comment by FANNG1** [comment]
@xunliu @yuqi1129 @tengqm are there any other comments?

**Review by xunliu — APPROVED** [review]
@SophieTech88 Thank you for your contributions.
LGTM

**Review by FANNG1 — APPROVED** [review]

**Pull request merged**